src/sagas.js

```diff
@@ -245,6 +282,80 @@ function* startVm (action) {
   yield stopProgress({ vmId: action.payload.vmId, name: 'start', result })
 }
 
+function* showEditVm (action) {
+  yield put(updateDialogType('edit'))
+  const cluster = Selectors.getClusterById(action.payload.vm.get('cluster').get('id'))
+  yield put(updateCluster(cluster))
+
+  const template = Selectors.getTemplateById(action.payload.vm.get('template').get('id'))
+  yield put(updateTemplate(template))
+
+  yield put(updateVmId(action.payload.vm.get('id')))
+
+  const os = Selectors.getOperatingSystemByName(action.payload.vm.get('os').get('type'))
+  yield put(updateOperatingSystem(os))
+
+  const name = action.payload.vm.get('name')
+  yield put(updateVmName(name))
+
+  const memory = action.payload.vm.get('memory').get('total')
+  yield put(updateVmMemory(memory))
+
+  const cpu = action.payload.vm.get('cpu').get('vCPUs')
+  yield put(updateVmCpu(cpu))
+  yield put(openVmDialog())
+  yield put(setVmDetailToShow({ vmId: action.payload.vm.get('id') }))
+}
+
+function* showAddNewVm (action) {
+  yield put(setVmDetailToShow({ vmId: '0' }))
+  yield put(updateDialogType('create'))
+  const cluster = Selectors.getFirstCluster()
+  yield put(changeCluster(cluster))
+  yield put(updateVmId('0'))
+  yield put(updateVmName(''))
+  yield put(openVmDialog())
+}
+
+function* handleClusterChange (action) {
+  yield put(updateCluster(action.payload.cluster))
+  // After every cluster change, set template to Blank
+  const blankTemplate = Selectors.getTemplateById('00000000-0000-0000-0000-000000000000')
+  yield put(changeTemplate(blankTemplate))
+}
+
+function* handleTemplateChange (action) {
```

Comment:
It would be maybe good to do this right now while the patch is still young and the dialog does not have too much properties. 
So, what I mean - imagine this scenario:
- you have a VM and it was a hard labor to get all it's properties tuned according to your need.
- it is working and all is fine
- than you get access to a new cluster with more powerful hosts or whatever so you want to move it there
- so, you open the edit VM dialog, change the cluster and all is gone. All your settings are reset to defaults

Here we need to be a bit more smart. Either distinguish between new and edit (what I don't like but it is what we do in oVirt) or be even more smart and be able to decide if this field has been modified by the user or did not.
For edit VM in initialization all fields would be marked as modified by user and for new none. Than if you touch on new some field, it will be marked as modified by user. Than, if you change the template for example, the particular value will be rewritten from template only if it does not make sense in the context of new template, otherwise will stay intact.

@mkrajnak @mareklibra @jniederm @vojtechszocs @matobet what you guys think?